Source/MQTTnet/Packets/MqttTopicFilter.cs

```diff
@@ -45,7 +45,15 @@ public sealed class MqttTopicFilter
         ///     separator).
         /// </summary>
         public string Topic { get; set; }
-
+        /// <summary>
+        /// Compares TopicFilters on Topic strings.  Overlapping Topics are not considered.
+        /// </summary>
+        public override bool Equals( object other )
```

Comment:
Instead of comparing the topic here only I recommend to only compare the topic where it is used in the other code. So that this Equals override is no longer needed. 

From MSDN (https://docs.microsoft.com/en-us/dotnet/api/system.object.equals?view=net-6.0):

> You should not override [Equals](https://docs.microsoft.com/en-us/dotnet/api/system.object.equals?view=net-6.0) on a mutable reference type. This is because overriding [Equals](https://docs.microsoft.com/en-us/dotnet/api/system.object.equals?view=net-6.0) requires that you also override the [GetHashCode](https://docs.microsoft.com/en-us/dotnet/api/system.object.gethashcode?view=net-6.0) method, as discussed in the previous section. This means that the hash code of an instance of a mutable reference type can change during its lifetime, which can cause the object to be lost in a hash table.

It is also dangerous in my opinion to just compare some of the properties and not all because the caller might not know this.